lib/libnetlink/netlink.c

```diff
@@ -390,53 +395,65 @@ int netl_listen(struct netl_handle *h, void *args)
 	fds[0].fd = h->fd;
 
 	while (h->closing != 1) {
-		int res = poll(fds, 1, NETL_POLL_TIMEOUT);
-		if (res < 0 && errno != EINTR) {
-			perror("error during cmdline_run poll");
-			return 0;
-		}
-		if (fds[0].revents & POLLIN) {
-			iov.iov_len = sizeof(buf);
-			status = recvmsg(h->fd, &msg, 0);
-			if (status < 0) {
-				// TODO: EINT / EAGAIN / ENOBUF should continue
-				return -1;
-			}
-
-			if (status == 0) {
-				// EOF
-				return -1;
-			}
-
-			if (msg.msg_namelen != sizeof(nladdr)) {
-				// Invalid length
-				return -1;
-			}
-
-			for (hdr = (struct nlmsghdr *) buf;
-				 (size_t) status >= sizeof(*hdr);) {
-				len = hdr->nlmsg_len;
-				buflen = len - sizeof(*hdr);
-
-				if (buflen < 0 || buflen > status) {
-					// truncated
-					return -1;
-				}
-
-				err = netl_handler(h, &nladdr, hdr, args);
-				if (err < 0)
-					return err;
-
-				status -= NLMSG_ALIGN(len);
-				hdr =
-					(struct nlmsghdr *) ((char *) hdr + NLMSG_ALIGN(len));
-			}
-
-			if (status) {
-				// content not read
-				return -1;
-			}
-		}
+    if (poll(fds, 1, NETL_POLL_TIMEOUT) > 0) {
```

Comment:
you're suppose to check errno, on return, everything else than errno EINTR is considered as fatal